### DISCUSSION TITLE
Form submit button customization methods

### DISCUSSION THREAD

**Discussion by studiowizjo** [filamentphp/filament#3006] [Help] [answered]
I got a quite long form, which I`m using dozens of times daily and it has some some fileds which not necessarily have to be completed. To quick up filling it would be nice if submit button could be fixed and visible all the time, without scrolling need.

It looks for me that currently it is not possible to change behaviour of submit in anyway (except making view with own button).

What do you think of adding some API (I can make PR) for this, eg.:
```php
public static function form(Form $form): Form
{
  return $form
    ->submitLabel('Process')
    ->submitExtraAttributes(['class' => 'fixed bottom-5 right-8'])
    ->submitAction(fn() => //...)
    ->schema([ //... ]);
}
```

**Comment by danharrin** [comment, marked as answer]
AFAIK you can do this already by extending `getCreateFormAction()` or `getSaveFormAction()` on resource pages.

```php
protected function getCreateFormAction(): Action
{
    return parent:getCreateFormAction()
        ->label('Process')
        ->extraAttributes(['class' => 'fixed bottom-5 right-8']);
}
```

**Reply by studiowizjo** [reply]
Perfect, I could not find this in docs and didn`t tracked either in core.

**Reply by MarJose123** [reply]
> AFAIK you can do this already by extending `getCreateFormAction()` or `getSaveFormAction()` on resource pages.
> 
> ```
> protected function getCreateFormAction(): Action
> {
>     return parent:getCreateFormAction()
>         ->label('Process')
>         ->extraAttributes(['class' => 'fixed bottom-5 right-8']);
> }
> ```

is this also available for custom page?

**Reply by danharrin** [reply]
We don't have submit buttons by default on custom pages?
